src/praisonai-code/praisonai_code/cli/interactive/async_tui.py

```
@@ -852,12 +879,28 @@ def _execute_prompt(self, prompt: str) -> Optional[str]:
                 return f"Error: {auth_error}"
 
-            return str(response) if response else None
+            # Surface explicit failures reported by wrapper objects even when
+            # no auth signature matched (e.g. AutonomyResult.success is False).
+            result_error = getattr(response, "error", None)
+            if result_error:
+                self._last_error = RuntimeError(str(result_error))
+                return f"Error: {result_error}"
+            if getattr(response, "success", None) is False:
+                self._last_error = RuntimeError("Agent execution failed")
+                return f"Error: Agent execution failed"
```

Ruff (0.15.20)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)